dimod/utilities.py

```diff
@@ -340,3 +345,48 @@ def resolve_label_conflict(mapping, old_labels=None, new_labels=None):
             # don't need to add it to intermediate_to_new because it is a self-label
 
     return old_to_intermediate, intermediate_to_new
+
+
+def child_structure_dfs(sampler, seen=None):
+    """Find the structure of a composed sampler by a depth first search on the
+    children.
```

Comment:
What if there are multiple structured children? And one structure is closer to root, and makes more sense, but DFS returns some other structure from a deeper child?

Reply:
There are lots of ways to resolve there being multiple structured children. It is up to the implementer to choose their algorithm. This one is very explicitly depth-first-search so they are opting into the behaviour.

Reply:
Sure, if each composite can choose the algorithm it uses, and this is only one strategy, all good.